api4/channel_local.go

```diff
@@ -206,3 +206,40 @@ func localPatchChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Write([]byte(rchannel.ToJson()))
 }
+
+func localDeleteChannel(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	channel, err := c.App.GetChannel(c.Params.ChannelId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("localDeleteChannel", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	auditRec.AddMeta("channeld", channel)
+
+	if channel.Type == model.CHANNEL_DIRECT || channel.Type == model.CHANNEL_GROUP {
+		c.Err = model.NewAppError("localDeleteChannel", "api.channel.delete_channel.type.invalid", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	if c.Params.Permanent {
+		err = c.App.PermanentDeleteChannel(channel)
```

Comment:
Why this is not respecting to `EnableAPIChannelDeletion`?

Reply:
@isacikgoz local mode is like super admin, so its not meant to respect this option.

Reply:
I see, I wonder if this should mentioned in docs or elsewhere. I would expect the local mode behavior is same with the authenticated system admin over tcp. Any ideas @thefactremains?

Reply:
I've added documentation PR for the similar `EnableAPITeamDeletion` option - https://github.com/mattermost/docs/pull/3767/

I'll open docs for this PR as well, after it is reviewed.

Reply:
I've opened PRs for docs:

https://github.com/mattermost/docs/pull/3825
https://github.com/mattermost/mattermost-api-reference/pull/563